packages/node_modules/@webex/plugin-meetings/src/meeting/util.js

```diff
@@ -673,45 +696,73 @@ MeetingUtil.setPeerConnectionEvents = (meeting) => {
         break;
       case ICE_STATE.COMPLETED:
         LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> ICE STATE COMPLETED.');
-        Metrics.postEvent({event: eventType.ICE_END, meeting});
         break;
       case ICE_STATE.CONNECTED:
+        Metrics.postEvent({event: eventType.ICE_END, meeting});
+        meeting.setNetworkStatus(NETWORK_STATUS.CONNECTED);
         meeting.reconnectionManager.iceReconnected();
         LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> ICE STATE CONNECTED.');
         break;
       case ICE_STATE.CLOSED:
         LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> ICE STATE CLOSED.');
         break;
       case ICE_STATE.DISCONNECTED:
+        meeting.setNetworkStatus(NETWORK_STATUS.DISCONNECTED);
         meeting.reconnectionManager.waitForIceReconnect()
           .catch(() => {
             LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> ICE STATE DISCONNECTED. Automatic Reconnection Timed Out.');
 
-            // TODO - uncomment this once meeting.reconnect() changes are implemented
-            // meeting.reconnect();
+            meeting.reconnect();
           });
-
         LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> ICE STATE DISCONNECTED.');
         break;
       case ICE_STATE.FAILED:
         LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> ICE STATE FAILED.');
         // notify of ice failure
-        Metrics.postEvent({
-          event: eventType.ICE_END,
-          meeting,
-          data: {
-            canProceed: false,
-            errors: [
-              Metrics.generateErrorPayload(
-                2004, false, error.name.MEDIA_ENGINE
-              )]
-          }
-        });
+        // Ice failure is the only indicator currently for identifying the actual connection drop
+        // Firefox takes sometime 10-15 seconds to go to failed state
+        connectionFailed();
+        break;
+      default:
+        break;
+    }
+  };
+
+  peerConnection.onconnectionstatechange = () => {
+    LoggerProxy.logger.log('Meeting:util#setPeerConnectionEvents --> CONNECTION STATE CHANGE.');
```

Comment:
Please change all these `.log`s to `.info`s. The consumer will not need this information.